package-drone/src/main/java/de/dentrassi/pm/jenkins/DroneRecorder.java

```diff
@@ -485,19 +433,40 @@ public Void invoke ( final File basedir, final VirtualChannel channel ) throws I
                             filename = f;
                         }
                         uploader.upload ( file, filename );
-//                        TODO this.artifacts.put ( artifacId, f );
                     }
-
+                    uploadResult.addArtifacts ( uploader.complete () );
```

Comment:
I think uploader.complete() should not be called when there is an exception in upload()
As of V3 complete is just another step in the upload process.
-> Introduced ` public Map<String, String> getUploadedArtifacts ();` in Uploader